peachjam/models/core_document_model.py

```diff
@@ -79,6 +81,54 @@ def update_languages(self):
             self.languages = langs
             self.save()
 
+    def update_extracted_citations(self):
+        """Update the current work's ExtractedCitations."""
+        related_works = Work.objects.filter(
+            frbr_uri__in=self.fetch_related_works_frbr_uris()
+        )
+
+        # delete existing extracted citations
+        ExtractedCitation.objects.filter(citing_work=self).delete()
+
+        for related_work in related_works:
+            ExtractedCitation.objects.get_or_create(
+                citing_work=self, target_work=related_work
+            )
+
+    def fetch_related_works_frbr_uris(self):
```

Comment:
In peachjam, a `related` has a special meaning. So perhaps `fetch_cited_works_frbr_uris` is better?